### PR TITLE
HPCC4J-553 Github Actions: Add baremetal test workflow

### PR DESCRIPTION
**Pull request by jpmcmu**
- Renamed existing k8s workflow
- Added baremetal workflow

Signed-off-by: James McMullan James.McMullan@lexisnexis.com

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [x] I have created a corresponding JIRA ticket for this submission
- [x] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [x] The change has been fully tested:
  - [ ] This change does not cause any existing JUnits to fail.
  - [ ] I have include JUnit coverage to test this change
  - [ ] I have performed system test and covered possible regressions and side effects.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] This change fixes the problem, not just the symptom

### CODE REVIEW

File: .github/workflows/baremetal-regression-suite.yml [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        python -m site
+        python -m pip install --upgrade pip setuptools wheel
+
+    - uses: actions/checkout@v3
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+        fetch-depth: 0
+
+    - name: Rebase
+      if: github.event_name == 'pull_request'
```
Comment (rpastrana) [inline .github/workflows/baremetal-regression-suite.yml:38]: does github provide a mechanism to save and re-use jobs? If so, it feels like we should reuse some of these

File: .github/workflows/baremetal-regression-suite.yml [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+                branchVersionMatch = candidateBranchPattern.match(branchName)
+                branchVersion = extractVersion(branchVersionMatch.group(1))
+
+                # Get latest release in branch
+                findLatestBranchVer = "git tag --list 'hpcc4j_" + str(branchVersion[0]) + "." + str(branchVersion[1]) + "*-release' --sort=-v:refname | head -n 1"
+                return getTagVersionForCmd(findLatestBranchVer)
+
+        branch_name = os.environ['BRANCH_NAME']
+        latestVersion = getLatestBranchVersion(branch_name)
+        latestVersionURL = 'https://cdn.hpccsystems.com/releases/CE-Candidate-' + latestVersion + '/bin/platform/hpccsystems-platform-community_' + latestVersion + '-1jammy_amd64_withsymbols.deb'
```
Comment (rpastrana) [inline .github/workflows/baremetal-regression-suite.yml:127]: we'll have to communicate this dependency to Clo, just so they know to let us know before they plan to move away from this cdn, etc.

File: .github/workflows/baremetal-regression-suite.yml [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+        # Return latest version
```
Comment (rpastrana) [inline .github/workflows/baremetal-regression-suite.yml:129]: leftover commented out code?

**Review by rpastrana — COMMENTED** [review]
@jpmcmu seems fine, left a few comments/questions

**Comment by jpmcmu** [comment]
@rpastrana Fixed the git tag issue and addressed code review comments. Please review

File: .github/workflows/baremetal-regression-suite.yml [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+                branchVersion = extractVersion(branchVersionMatch.group(1))
+
+                # Get latest release in branch
+                findLatestBranchVer = "git tag --list 'hpcc4j_" + str(branchVersion[0]) + "." + str(branchVersion[1]) + "*-release' --sort=-v:refname | head -n 1"
+                return getTagVersionForCmd(findLatestBranchVer)
+
+        branch_name = os.environ['BRANCH_NAME']
+
+        latestVersion = getLatestBranchVersion(branch_name)
+        previousVersion = [latestVersion[0], latestVersion[1], latestVersion[2] - 2]
```
Comment (rpastrana) [inline .github/workflows/baremetal-regression-suite.yml:115]: We should report failure if point==0 and cannot be found

**Review by rpastrana — COMMENTED** [review]
@jpmcmu one comment

**Review by rpastrana — APPROVED** [review]

**Pull request merged**